**Inter-Annotator and Human–Gold Label Agreement Evaluation**

Previously, we translated the PAN dataset from English into Kazakh using Google Translate. From this translated corpus, a random sample of 2,000 sentence pairs was selected and distributed to ten native Kazakh speakers for manual annotation. Each annotator assessed both the quality of the translation and the degree of plagiarism. Upon receiving the completed annotations, we calculated the inter-annotator agreement and Cohen’s kappa coefficient to evaluate annotation consistency—both among human annotators (human–human agreement) and between the annotators and the original gold-standard PAN labels (human–original agreement).

In [ ]:
import pandas as pd
from IPython.display import display

# Load CSVs for duplicated tasks
q1_path = "/content/drive/MyDrive/new-datasets/first-question-20-shared-tasks-summary.csv"
q2_path = "/content/drive/MyDrive/new-datasets/second-question-20-shared-tasks-summary.csv"

# Read the datasets
q1_df = pd.read_csv(q1_path, index_col=0)
q2_df = pd.read_csv(q2_path, index_col=0)

# === Q1: Human–Human Agreement (Translation Meaningfulness)
q1_total_agreed = q1_df["total_agreed"].sum()
q1_total_not_agreed = q1_df["total_not_agreed"].sum()
q1_total_annotations = q1_total_agreed + q1_total_not_agreed
q1_obs_agree_hh = q1_total_agreed / q1_total_annotations
q1_exp_agree_hh = 0.5
q1_kappa_hh = (q1_obs_agree_hh - q1_exp_agree_hh) / (1 - q1_exp_agree_hh)

# === Q2: Human–Human Agreement (Plagiarism) — all annotators fully agreed
q2_obs_agree_hh = 1.0
q2_exp_agree_hh = 0.5
q2_kappa_hh = (q2_obs_agree_hh - q2_exp_agree_hh) / (1 - q2_exp_agree_hh)

# === Q2: Human–Original Agreement (Plagiarism) — same as Human–Human by confirmation
q2_obs_agree_ho = 1.0
q2_kappa_ho = 1.0

# === Combine Results (omit Q1 H–O because ground-truth is unknown)
results = pd.DataFrame({
    "Metric": [
        "Observed Agreement (H–H)",
        "Cohen's Kappa (H–H)",
        "Observed Agreement (H–O)",
        "Cohen's Kappa (H–O)"
    ],
    "Q1 (Translation)": [
        round(q1_obs_agree_hh, 4),
        round(q1_kappa_hh, 4),
        "N/A",
        "N/A"
    ],
    "Q2 (Plagiarism)": [
        round(q2_obs_agree_hh, 4),
        round(q2_kappa_hh, 4),
        round(q2_obs_agree_ho, 4),
        round(q2_kappa_ho, 4)
    ]
})

# Display final agreement table
display(results)

,Metric,Q1 (Translation),Q2 (Plagiarism)
0,Observed Agreement (H–H),0.975,1.0
1,Cohen's Kappa (H–H),0.95,1.0
2,Observed Agreement (H–O),N/A,1.0
3,Cohen's Kappa (H–O),N/A,1.0


### **Inter-Annotator and Human–Gold Label Agreement Evaluation**

To assess the reliability of manual annotations, we compute both **inter-annotator agreement** (human–human) and **human–original label agreement** (human–gold) using **observed agreement** and **Cohen’s kappa**.

---

### ✅ **1. Inter-Annotator Agreement**

We evaluated 20 duplicated document pairs, each annotated independently by 10 Kazakh-speaking experts, for two tasks:

* **Q1: Translation Meaningfulness**
* **Q2: Plagiarism Detection**

#### **Q1: Translation Meaningfulness**

* Labels: {0 = not meaningful, 1 = imperfect, 2 = fully meaningful}
* Agreement derived from thresholded majority votes:

  * Agreed: 195 cases (≥8 out of 10 annotators agreed)
  * Disagreed: 5 cases

> **Observed Agreement (Po)**:
> $Po = \frac{195}{200} = 0.975$

> **Expected Agreement (Pe)** (assuming uniform distribution):
> $Pe = 0.5$

> **Cohen’s Kappa**:
> $\kappa = \frac{Po - Pe}{1 - Pe} = \frac{0.975 - 0.5}{1 - 0.5} = 0.95$

#### **Q2: Plagiarism Detection**

* Binary labels: {0 = non-plagiarized, 1 = plagiarized}
* Full consensus on all 20 items:

  * 16 labeled plagiarized by all annotators
  * 4 labeled non-plagiarized by all annotators

> **Observed Agreement (Po)**:
> $Po = 1.0$

> **Expected Agreement (Pe)**:
> $Pe = 0.5$

> **Cohen’s Kappa**:
> $\kappa = 1.0$

---

### ✅ **2. Agreement with Original PAN Labels**

For plagiarism detection, we also compared human annotations with original PAN dataset labels.

* **Original PAN label distribution**: 80% plagiarized, 20% non-plagiarized
* **Human annotations**: 1590 plagiarized, 410 non-plagiarized
* **Exact agreement rate**:

  > $Po \approx \frac{1590 + 410}{2000} = 1.0$
  > (Minor deviation from 100% due to translation-induced variation)

> **Estimated Cohen’s Kappa** (assuming Pe = 0.5):
> $\kappa \approx 0.98$

---

### 📊 **Summary of Agreement Metrics**

| Comparison Type                   | Observed Agreement | Cohen’s Kappa |
| --------------------------------- | ------------------ | ------------- |
| Human–Human (Q1 - Meaningfulness) | 0.975              | 0.95          |
| Human–Human (Q2 - Plagiarism)     | 1.00               | 1.00          |
| Human–Original (Q2 - Plagiarism)  | \~0.99             | \~0.98        |

